json-endpoints/class.wpcom-json-api-update-post-v1-1-endpoint.php

```diff
@@ -330,20 +332,34 @@ function write_post( $path, $blog_id, $post_id ) {
 		$has_media = ! empty( $input['media'] ) ? count( $input['media'] ) : false;
 		$has_media_by_url = ! empty( $input['media_urls'] ) ? count( $input['media_urls'] ) : false;
 
-		if ( $new ) {
+		$media_id_string = '';
+		if ( $has_media || $has_media_by_url ) {
+			$media_files = ! empty( $input['media'] ) ? $input['media'] : array();
+			$media_urls = ! empty( $input['media_urls'] ) ? $input['media_urls'] : array();
+			$media_attrs = ! empty( $input['media_attrs'] ) ? $input['media_attrs'] : array();
+			$media_results = $this->handle_media_creation_v1_1( $media_files, $media_urls, $media_attrs );
+			$media_id_string = join( ',', array_filter( array_map( 'absint', $media_results['media_ids'] ) ) );
+		}
 
+		if ( $new ) {
 			if ( isset( $input['content'] ) && ! has_shortcode( $input['content'], 'gallery' ) && ( $has_media || $has_media_by_url ) ) {
 				switch ( ( $has_media + $has_media_by_url ) ) {
 				case 0 :
 					// No images - do nothing.
 					break;
 				case 1 :
 					// 1 image - make it big
-					$insert['post_content'] = $input['content'] = "[gallery size=full columns=1]\n\n" . $input['content'];
+					$insert['post_content'] = $input['content'] = sprintf(
+						"[gallery size=full ids='%s' columns=1]\n\n",
+						$media_id_string
```

Comment:
This should only ever be a comma seperated list of integers, so I'd be happier if we could run a preg_match() on it to ensure it's only `^[\d\,]*$` or something